src/SignalR/perf/benchmarkapps/Crankier/Server/ConnectionCounter.cs

```diff
@@ -0,0 +1,56 @@
+using System;
+
+namespace Microsoft.AspNetCore.SignalR.Crankier.Server
+{
+    public class ConnectionCounter
+    {
+        private int _totalConnectedCount;
+        private int _peakConnectedCount;
+        private int _totalDisconnectedCount;
+        private int _receivedCount;
+
+        private readonly object _lock = new object();
+
+        public ConnectionSummary Summary
+        {
+            get
+            {
+                lock (_lock)
+                {
+                    return new ConnectionSummary
+                    {
+                        CurrentConnections = _totalConnectedCount - _totalDisconnectedCount,
+                        PeakConnections = _peakConnectedCount,
+                        TotalConnected = _totalConnectedCount,
+                        TotalDisconnected = _totalDisconnectedCount,
+                        ReceivedCount = _receivedCount
+                    };
+                }
+            }
+        }
+
+        public void Receive(string a) {
```

Comment:
nit: newline for `{`

Reply:
Is this used for anything?

Reply:
I wired it back up to `EchoHub\SendPayload`.  It doesn't echo yet (still thinking of a way to not have it look ugly), and Crankier itself doesn't call `SendPayload` yet directly, i'll open a PR with this at a later date.